src/applications/model/lognorm3-random-variable.cc

```diff
@@ -0,0 +1,91 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#include "ns3/log.h"
+#include "ns3/assert.h"
+#include "ns3/simulator.h"
+#include "lognorm3-random-variable.h"
+#include "ns3/double.h"
+
+namespace ns3 {
+
+NS_LOG_COMPONENT_DEFINE ("LogNormal3RandomVariable");
+
+NS_OBJECT_ENSURE_REGISTERED(LogNormal3RandomVariable);
+
+TypeId
+LogNormal3RandomVariable::GetTypeId (void)
+{
+  static TypeId tid = TypeId ("ns3::LogNormal3RandomVariable")
+      .SetParent<LogNormalRandomVariable> ()
+      .SetGroupName ("GamingTraffic")
+      .AddConstructor<LogNormal3RandomVariable> ()
+      .AddAttribute("Threshold", "The threshold value for the three-parameters log-normal distribution returned by this RNG stream.",
+                    DoubleValue (0.0),
+                    MakeDoubleAccessor (&LogNormal3RandomVariable::m_thresh),
+                    MakeDoubleChecker<double> ())
+  ;
+  return tid;
+}
+
+LogNormal3RandomVariable::LogNormal3RandomVariable ()
+{
+  NS_LOG_FUNCTION (this);
+}
+
+double
+LogNormal3RandomVariable::GetThreshold (void) const
+{
+  NS_LOG_FUNCTION (this);
+  return m_thresh;
+}
+
+double
+LogNormal3RandomVariable::GetValue (double threshold)
+{
+  NS_LOG_FUNCTION (this << threshold);
+  double x = LogNormalRandomVariable::GetValue ();
+  double y = x + threshold;
+  return y;
+}
+
+uint32_t
+LogNormal3RandomVariable::GetInteger (uint32_t threshold)
```

Comment:
Shouldn't the threshold value be a double?

Reply:
I don't the reason behind it, but in other RandomVariables, they used `uint32_t` for parameters of `GetInteger`. I think it's better to follow the other variables.

Reply:
Maybe I'm wrong, let me know

Reply:
We have to `GetInteger` method with different signatures, first one without parameter and the second one with `uint32_t` parameters just like as other RandomVariables.
both of them are used to generate the Integer numbers. but I don't know the difference.

Reply:
Well the difference is that you can pass different parameters instead of the default class members to compute the value to return.
Regarding the fact that threshold is uint when calling GetInteger, this must be kept in mind since if we pass a double value it will be rounded to an int, leading to a possible unwanted output value

Reply:
Here we never use the `GetInteger` with the uint32. In case we want to generate integer numbers, we call the GetInteger without parameter.

I just GetInteger with uint to adopt the style used in other RandomVariables.